crates/integration-tests/src/tests/run_ephemeral_ssh.rs

```diff
@@ -358,3 +358,67 @@ fn test_run_ephemeral_ssh_broken_image_cleanup() -> Result<()> {
     Ok(())
 }
 integration_test!(test_run_ephemeral_ssh_broken_image_cleanup);
+
+/// Test ephemeral VM network and DNS
+///
+/// Verifies that ephemeral bootc VMs can access the network and resolve DNS correctly.
+/// 1. Checks IP connectivity via ping or TCP fallback.
+/// 2. Verifies DNS resolution for a public host (google.com).
+fn test_run_ephemeral_dns_resolution() -> Result<()> {
+    // Check basic IP connectivity (ping or TCP fallback)
+    let ip_test = run_bcvk(&[
+        "ephemeral",
+        "run-ssh",
+        "--label",
+        INTEGRATION_TEST_LABEL,
+        &get_test_image(),
+        "--",
+        "/bin/sh",
+        "-c",
+        r#"
+        for i in $(seq 1 30); do
+            ip -4 addr show | grep -q "inet " && break
+            sleep 1
+        done
+        ping -c1 -W3 1.1.1.1 >/dev/null 2>&1 || \
+        ping -c1 -W3 8.8.8.8 >/dev/null 2>&1 || \
+        timeout 2 bash -c 'echo > /dev/tcp/1.1.1.1/53' 2>/dev/null || \
+        timeout 2 bash -c 'echo > /dev/tcp/8.8.8.8/80' 2>/dev/null || exit 1
+        echo "IP connectivity: OK"
+        "#,
+    ])?;
+
+    assert!(ip_test.success() && ip_test.stdout.contains("IP connectivity: OK"));
+
+    // Test DNS resolution
+    let dns_output = run_bcvk(&[
+        "ephemeral",
+        "run-ssh",
+        "--label",
+        INTEGRATION_TEST_LABEL,
+        &get_test_image(),
+        "--",
+        "/bin/sh",
+        "-c",
+        r#"getent hosts google.com || ping -c1 -W2 google.com | head -1"#,
+    ])?;
+
+    let out = dns_output.stdout;
+    let resolved = out.contains("google.com")
+        || out.contains("PING")
+        || out.contains("2001:")
+        || out.contains("142.")
+        || out.contains("104.")
+        || out.contains("108.")
+        || out.contains("172.217.")
+        || out.contains("216.58.");
```

Comment:
This list of hardcoded IP addresses seems a bit weird...why not just require `getent hosts` to work?

Or actually building on the above, how about just `podman pull quay.io/fedora/fedora` or so for a more realistic networking test?